ote/src/clj/ote/services/index.clj

```diff
@@ -68,8 +70,10 @@
             window['ga-disable-" (:tracking-code ga-conf) "'] = true;
           }")))
 
-      [:body {:onload        "ote.main.main();"
-             :data-language localization/*language*}
+      [:body {:id           "main-body"
+              :onload       "ote.main.main();"
+              :features     flags
```

Comment:
I find it strange that the feature flags are injected into html attribute as a string, whereas you could inject them as default values in the app state tree. 
If the flags would be in the state, we could also easily toggle them on/off in the browser console using some utility function, and test them out in live situation. This makes also sure that the flagged features are properly rendered if the state changes.

Reply:
That would work too if we want to add feature flags to code. 

Currently we can't change app state without building new .jar file to the server. As long as feature flags are in config file we can change config and restart server. No new build is required. 

So we don't gain anything by changing the current implementation. It would only be different. And those values are just as easily changeable to the dom as app state is. JS code is required. In both situations.

Maybe the best option would be database based configs.

Reply:
This works, so it is ok. But, this feels like a "karvalakki" approach. Passing flags through state or react props would be cleaner, but that is not critical. I'll stop my ranting now :)

Reply:
Features often have implications on both the server and the client side, so they need to be passed in the initial render somehow. Attribute is a good enough solution that is easy to read in the client side without needing to make an AJAX call the get the flags config.